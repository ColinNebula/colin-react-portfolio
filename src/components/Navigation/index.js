import React, { useState, useEffect } from "react";
import Navbar from 'react-bootstrap/Navbar';
import { Container } from 'react-bootstrap';
import logoM from '../../assets/images/logoM.png';
import Nav from 'react-bootstrap/Nav';
import { FaUser, FaBriefcase, FaFileAlt, FaEnvelope, FaMoon, FaSun } from 'react-icons/fa';

function CustomNav(props) {
    const { currentTab, setCurrentTab } = props;
    const [scrolled, setScrolled] = useState(false);
    const [darkMode, setDarkMode] = useState(false);

    // Handle navbar background on scroll
    useEffect(() => {
        const handleScroll = () => {
            const isScrolled = window.scrollY > 50;
            setScrolled(isScrolled);
        };

        window.addEventListener('scroll', handleScroll);
        return () => window.removeEventListener('scroll', handleScroll);
    }, []);

    const navItems = [
        { key: 'about', label: 'About', icon: FaUser },
        { key: 'portfolio', label: 'Portfolio', icon: FaBriefcase },
        { key: 'resume', label: 'Resume', icon: FaFileAlt },
        { key: 'contact', label: 'Contact', icon: FaEnvelope }
    ];

    const toggleDarkMode = () => {
        setDarkMode(!darkMode);
        // Add dark mode logic here if needed
    };

    return (
        <Navbar 
            className={`enhanced-navbar ${scrolled ? 'scrolled' : ''} ${darkMode ? 'dark-mode' : ''}`}
            expand="lg" 
            fixed="top"
            collapseOnSelect
        >
            <Container>
                <Navbar.Brand 
                    href="#home" 
                    className="brand-enhanced"
                    onClick={() => setCurrentTab("about")}
                >
                    <div className="brand-container">
                        <img 
                            src={logoM} 
                            className="brand-logo" 
                            alt="Colin Nebula Logo" 
                        />
                        <div className="brand-text">
                            <span className="brand-name">Colin Nebula</span>
                            <span className="brand-tagline">Full Stack Developer</span>
                        </div>
                    </div>
                </Navbar.Brand>

                <Navbar.Toggle 
                    aria-controls="enhanced-navbar-nav" 
                    className="navbar-toggler-enhanced"
                >
                    <span></span>
                    <span></span>
                    <span></span>
                </Navbar.Toggle>

                <Navbar.Collapse id="enhanced-navbar-nav">
                    <Nav className="ms-auto nav-enhanced">
                        {navItems.map((item) => (
                            <Nav.Link 
                                key={item.key}
                                className={`nav-item-enhanced ${currentTab === item.key ? "active" : ""}`}
                                onClick={() => setCurrentTab(item.key)}
                            >
                                <item.icon className="nav-icon-enhanced" />
                                <span className="nav-label">{item.label}</span>
                                <div className="nav-indicator"></div>
                            </Nav.Link>
                        ))}
                        
                        <div className="nav-divider"></div>
                        
                        <button 
                            className="theme-toggle"
                            onClick={toggleDarkMode}
                            title={darkMode ? "Switch to Light Mode" : "Switch to Dark Mode"}
                        >
                            {darkMode ? <FaSun /> : <FaMoon />}
                        </button>
                    </Nav>
                </Navbar.Collapse>
            </Container>
        </Navbar>
    );
}

export default CustomNav;