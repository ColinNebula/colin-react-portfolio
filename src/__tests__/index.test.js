// __tests__/Nav.test.js with hard coded categories
import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Navigation from '../components/Navigation';

afterEach(cleanup);

describe('Navigation component', () => {
  it('renders', () => {
    render(<Navigation />);
  });

  it('matches snapshot', () => {
    const { asFragment } = render(<Navigation />);
    
    expect(asFragment()).toMatchSnapshot();
  });
})

describe('emoji is visible', () => {
  it('inserts emoji into the h2', () => {
  const { getByLabelText } = render(<Navigation />);

  expect( getByLabelText('camera')).toHaveTextContent('ðŸ“¸');
  });
})   

describe('links are visible', () => {
  it('inserts text into the links', () => {
    const { getByTestId } = render(<Navigation />);

    expect(getByTestId('link')).toHaveTextContent('Colin Nebula!');
    expect(getByTestId('about')).toHaveTextContent('About');
  });

})