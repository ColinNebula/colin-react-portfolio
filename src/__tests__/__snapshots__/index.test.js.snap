// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav component matches snapshot 1`] = `
<DocumentFragment>
  <header>
    <nav
      class="navbar navbar-expand navbar-dark bg-dark"
    >
      <div
        class="container"
      >
        <a
          class="navbar-brand"
          href="/"
        >
          Colin Nebula 
        </a>
        <div
          class="navbar-collapse collapse"
        >
          <div
            class="me-auto navbar-nav"
          >
            <a
              class="nav-link"
              role="button"
              tabindex="0"
            >
               Contact 
            </a>
            <a
              class="nav-link"
              role="button"
              tabindex="0"
            >
               About Me 
            </a>
            <a
              class="nav-link"
              role="button"
              tabindex="0"
            >
               Portfolio 
            </a>
            <a
              class="nav-link"
              role="button"
              tabindex="0"
            >
               Resume 
            </a>
          </div>
        </div>
      </div>
    </nav>
  </header>
</DocumentFragment>
`;

exports[`Navigation component matches snapshot 1`] = `
<DocumentFragment>
  <nav>
    <ul
      class="flex-row"
    >
      <li
        class="mx-2"
      >
        <span>
          About
        </span>
      </li>
      <li
        class="mx-2"
      >
        <span>
          Portfolio
        </span>
      </li>
      <li
        class="mx-2"
      >
        <span>
          Resume
        </span>
      </li>
    </ul>
  </nav>
</DocumentFragment>
`;
